kyuubi-server/src/main/scala/org/apache/kyuubi/engine/spark/SparkProcessBuilder.scala

```diff
@@ -193,28 +194,32 @@ class SparkProcessBuilder(
     }
   }
 
-  override def killApplication(line: String = lastRowsOfLog.toArray.mkString("\n")): String =
-    YARN_APP_NAME_REGEX.findFirstIn(line) match {
-      case Some(appId) =>
-        try {
-          val hadoopConf = KyuubiHadoopUtils.newHadoopConf(conf)
-          yarnClient.init(hadoopConf)
-          yarnClient.start()
-          val applicationId = ApplicationId.fromString(appId)
-          yarnClient.killApplication(applicationId)
-          s"Killed Application $appId successfully."
-        } catch {
-          case e: Throwable =>
-            s"Failed to kill Application $appId, please kill it manually." +
-              s" Caused by ${e.getMessage}."
-        } finally {
-          if (yarnClient != null) {
-            yarnClient.stop()
-          }
+  override def killApplication(engineRefId: String): String = {
+    val master = conf.getOption("spark.master").getOrElse("local")
```

Comment:
In this way, `spark.master` can only be configured in `kyuubi-defaults.conf` to support automatic kill yarn app, but not in `spark-defaults.conf`?

nit:`spark.master` can be used as a constant of `object SparkProcessBuilder`.

Reply:
Yes, we can't get the configuration in spark-defaults.conf, or is there a way I don't know?
cc @yaooqinn

Reply:
We can get spark.master conf here; or else from spark-defaults.conf from "SPARK_CONF_DIR"; or else SPARK_HOME/conf

Reply:
FYI:
```
  private def getSparkDefaultsConf(): Map[String, String] = {
    val sparkDefaultsConfFile = env.get("SPARK_CONF_DIR")
      .orElse(env.get("SPARK_HOME").map(_ + "/conf"))
      .map(_ + "/spark-defaults.conf").map(new File(_))
      .filter(_.exists())
    Utils.getPropertiesFromFile(sparkDefaultsConfFile)
  }

conf.getOption("spark.master").getOrElse(getSparkDefaultsConf().get("spark.master"))
```

Reply:
thanks， I saw your pr is drafted，I’ll add it to this pr